torch/csrc/jit/python/init.cpp

```diff
@@ -148,6 +156,20 @@ static c10::SymFloatNode toSymFloatNode(c10::SymFloatNode a, py::object b) {
   }
 }
 
+// precondition: b is either a SymIntNode or an int
+static c10::SymIntNode toSymIntNode(c10::SymFloatNode a, py::object b) {
+  return torch::is_symint_node(b) ? b.cast<c10::SymIntNode>()
+                                  : a->sym_int()->wrap(b.cast<int64_t>());
+}
+
+// precondition: b is either a SymFloatNode or a double
+static c10::SymFloatNode toSymFloatNode(c10::SymIntNode a, py::object b) {
+  if (torch::is_symfloat_node(b)) {
+    return b.cast<c10::SymFloatNode>();
+  } else {
+    return a->sym_float()->wrap(b.cast<double>());
```

Comment:
Major change compared to what is in the branch: We have a `a->sym_float()` here instead of just `a`. If we don't do that, we end up with a `SymFloatNode(PySymInt(double))` which is... not good. Added extra assert in in PySym* to reduce the chance of these things happening.